### PR TITLE
LRQA-68009 Adopt SACRED model on SAML test cases and macros - A

### PR DESCRIPTION
**Pull request by gaborlovasdotcom**
https://issues.liferay.com/browse/LRQA-68009

### CODE REVIEW

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml

**Comment by gaborlovasdotcom** [comment]
ci:test:saml

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

File: portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro [hunk 1]
```diff
@@ -167,18 +298,92 @@ definition {
 			spNameID = "${spEntityID}");
 	}
 
+	@description = "Disable the Enable Identity Provider Role To Be Configured capability at System Settings"
+	macro disableIDRoleConfigurationAtSystemSettings {
```
Comment (ferenconodiD) [inline portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro:302]: a P is missing, should be: disableIDPRoleConfigurationAtSystemSettings

File: portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro [hunk 2]
```diff
@@ -1,5 +1,132 @@
...
+
+		SAMLNavigation.gotoIdentityProviderConnections();
+
+		SAMLPortlet.addIdentityProviderConnection(
+			idpAttributeMapping = "${idpAttributeMapping}",
+			idpEntityID = "${idpEntityID}",
+			idpKeepAliveURL = "${idpKeepAliveURL}",
+			idpMetadataURL = "${idpURL}/c/portal/saml/metadata",
+			idpNameID = "${idpEntityID}",
+			index = "${index}",
```
Comment (ferenconodiD) [inline portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro:33]: The name index could be more specific.

**Review by ferenconodiD — COMMENTED** [review]
Please take a look on my review remarks and consider them!

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

File: portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro [hunk 3]
```diff
@@ -193,18 +398,94 @@ definition {
...
+	@description = "Navigate to the SAMLAdmin on a logged in Instance"
+	macro navigateToSAMLAdmin {
+		ApplicationsMenu.gotoPortlet(
+			category = "Security",
+			panel = "Control Panel",
+			portlet = "SAML Admin");
+	}
+
+	@description = "7.4-SAML-S1: Configure Portal Instance to act as a SAML Identity Provider"
+	macro setupIDP {
```
[the remark this answers is not in the document]
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro:430]: Done

File: portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro [hunk 4]
```diff
@@ -193,18 +398,94 @@ definition {
...
+
+		// Logout from identity provider site
+
+		User.logoutPG(specificURL = "${idpURL}");
+
+		User.viewLoggedOutPG();
+	}
+
+	@description = "7.4-SAML-S2: Configure Portal Instance to act as a SAML Service Provider"
+	macro setupSP {
```
[the remark this answers is not in the document]
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro:446]: Done

File: portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro (hunk 2 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro:33]: Renamed to: samlAttributeIDforUserMatching

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
SACRED changes are lead by https://issues.liferay.com/browse/LRQA-67999

**Pull request closed without merging**